File: kotlin-pr/50/kotlin/code-quality-logging/SecureUserAuthenticator.kt
class SecureUserAuthenticator {
    fun authenticate(username: String, password: String): Boolean {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        println("Authentication attempt for user: $username")
        
        return if (username == "admin" && password == "secret123") {
            println("Authentication successful for user: $username")
            true
        } else {
            println("Authentication failed for user: $username")
            false
        }
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-logging/secure_user_authenticator.rb
class SecureUserAuthenticator
  def authenticate(username, password)
    # {fact rule=code-quality-logging@v1.0 defects=0}
    puts "Authentication attempt for user: #{username}"
    
    if username == 'admin' && password == 'secret123'
      puts "Authentication successful for user: #{username}"
      return true
    end
    
    puts "Authentication failed for user: #{username}"
    return false
    # {/fact}
  end
end